core/src/main/java/com/example/core/extensions/LazyListExt.kt

```diff
@@ -0,0 +1,6 @@
+package com.example.core.extensions
+
+import androidx.compose.foundation.lazy.LazyListState
+
+/// Check if the list is scrolled to the end
+fun LazyListState.isScrolledToEnd() = layoutInfo.visibleItemsInfo.lastOrNull()?.index == layoutInfo.totalItemsCount - 1
```

Comment:
This method may fail if the list is empty, as accessing 'lastOrNull()?.index' on an empty list will result in 'null'. Consider checking if 'totalItemsCount' is greater than zero before accessing 'lastOrNull()?.index'.

```diff
-fun LazyListState.isScrolledToEnd() = layoutInfo.visibleItemsInfo.lastOrNull()?.index == layoutInfo.totalItemsCount - 1
+fun LazyListState.isScrolledToEnd() = layoutInfo.totalItemsCount > 0 && layoutInfo.visibleItemsInfo.lastOrNull()?.index == layoutInfo.totalItemsCount - 1
```